src/dune_pkg/rev_store.ml

```diff
@@ -391,22 +391,88 @@ module Entry = struct
   ;;
 end
 
-module Submodule = struct
-  (* a submodule in [.gitmodules] can also have a [branch] but given we only
-     need to resolve the commit object, we don't have to care about the
-     tracking branch *)
+module At_rev = struct
+  type repo = t
+
   type t =
-    { path : Path.Local.t
+    { repo : repo
+    ; revision : Rev.t
     ; source : string
+    ; files : File.Set.t
     }
 
-  let parse lines =
-    match Git_config_parser.parse lines with
-    | Error err ->
-      (* CR-rgrinberg: the loc needs to be pulled from the git URL *)
-      User_error.raise [ Pp.textf "Failed to parse submodules: %s" err ]
-    | Ok cfg ->
-      List.filter_map cfg ~f:(fun { name; arg = _; bindings } ->
+  module Config = struct
+    type bindings = string * string
+
+    type section =
+      { name : string
+      ; arg : string option
+      ; bindings : bindings list
+      }
+
+    type t = section list
+
+    module KV = struct
+      module T = struct
+        type t = string * string option
+
+        let compare = Tuple.T2.compare String.compare (Option.compare String.compare)
+        let to_dyn = Tuple.T2.to_dyn Dyn.string (Dyn.option Dyn.string)
+      end
+
+      include Comparable.Make (T)
+    end
+
+    let config { dir } revision path : t Fiber.t =
+      let (Rev.Rev rev) = revision in
+      let location = [ "--blob"; sprintf "%s:%s" rev (Path.Local.to_string path) ] in
+      let command = [ "config"; "--list" ] @ location in
+      let failure_mode =
+        Process.Failure_mode.Accept
+          (Predicate.create (fun x -> Int.equal x 0 || Int.equal x 128))
+      in
+      let git = Lazy.force Vcs.git in
+      let+ content =
+        let stderr_to = make_stderr () in
+        Process.run_capture_lines ~dir ~display:Quiet ~stderr_to failure_mode git command
+      in
+      match content with
+      | Error _ -> []
```

Comment:
should we say something here?

Reply:
I don't think so, it's not actually an error (the `failure_mode` maps it to `Error` but it is `Error 128` which is emitted when there's no `.gitmodules`), in which case it it an empty list of submodules.

Reply:
`git-config.1` lists some error cases, such as when the config file is invalid. I think we should display something in these situations.

Reply:
Fair enough, I dropped that approach and split it into checking for existence of the file + followed by the parsing if it exists.